Gifski/MainWindowController.swift

```diff
@@ -27,20 +26,45 @@ final class MainWindowController: NSWindowController {
 		$0.centerInWindow(window)
 	}
 
+	private lazy var cancelButton = with(CustomButton.circularButton(title: "╳", size: 130)) {
+		$0.textColor = .appTheme
+		$0.backgroundColor = NSColor.appTheme.with(alpha: 0.1)
```

Comment:
That looks absolutely horrendous. The alpha prevents the background color from automatically updating. Should've checked that.

I agree with the `updateLayer` closure being a suitable workaround.

Reply:
I really wish it was possible to create dynamic colors in user-land... So `NSColor.appTheme.with(alpha: 0.1)` returned a NSColor instance that updated itself, exactly as `NSColor.appTheme` would.

Reply:
I went with a slightly different solution. Instead of providing an `onUpdateLayer` callback (that could cause some trouble with having to update properties that would in turn call `updateLayer` again), I've added a mechanism that allows you to set a 'color generator' for any of the key paths that return `NSColor`.

```swift
$0.setColorGenerator(for: \.backgroundColor) {
	NSColor.appTheme.with(alpha: 0.1)
}
```

The generator gets called in `animateColor` through a convenience method that returns either the generated color if it exists, or the standard property:

```swift
let backgroundColor = isOn ? color(for: \.activeBackgroundColor) : color(for: \.backgroundColor)
```